src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/service/IncentivesService.kt

```diff
@@ -1,27 +1,64 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.service
 
+import com.microsoft.applicationinsights.TelemetryClient
+import org.slf4j.LoggerFactory
 import org.springframework.data.domain.Page
 import org.springframework.data.domain.Pageable
 import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 import org.springframework.transaction.annotation.Transactional
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CodeDescription
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateIncentiveRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.CreateIncentiveResponse
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.IncentiveIdResponse
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.IncentiveResponse
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.filter.IncentiveFilter
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IEPLevel
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.Incentive
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.IncentiveId
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderBooking
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.AgencyLocationRepository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.IncentiveRepository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.specification.IncentiveSpecification
 import java.time.LocalDateTime
 
 @Service
 @Transactional
 class IncentivesService(
   private val incentiveRepository: IncentiveRepository,
-  private val offenderBookingRepository: OffenderBookingRepository
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val agencyLocationRepository: AgencyLocationRepository,
+  private val iepLevelRepository: ReferenceCodeRepository<IEPLevel>,
+  private val telemetryClient: TelemetryClient,
 ) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun createIncentive(bookingId: Long, dto: CreateIncentiveRequest): CreateIncentiveResponse {
+    val offenderBooking = offenderBookingRepository.findById(bookingId)
+      .orElseThrow(NotFoundException(bookingId.toString()))
+
+    val mappedIncentive = mapIncentiveModel(dto, offenderBooking)
+    val incentive = incentiveRepository.save(mappedIncentive)
```

Comment:
not sure why we need to save here? Can't we just add to booking and let the transaction commit?

Reply:
good point